geoips/pydantic_models/v1/bases.py

```diff
@@ -64,49 +122,128 @@ def __str__(self) -> str:
+        Parameters
+        ----------
+        values: dict
+            dictionary containing step details.
 
-    This model extends ``PrettyBaseModel`` and uses Pydantic's ConfigDict to provide
-    customized configurations. It is intended for use in cases where additional fields
-    are not allowed, and the object data cannot be modified after initialization.
+        Returns
+        -------
+        dict
+            dictionary containing step details.
+
+        Raises
+        ------
+        ValueError
+            If a restricted field is found in the input.
+
+        Notes
+        -----
+            This function was initially implemented to handle a restricted field called
+            `plugin_type`, which was removed as part of of architectural changes.
+            Although it is currently unused, we anticipate a strong use case moving
+            forward.
+        """
+        for field in cls.restricted_fields:
+            # if field in values and values[field] is not None:
+            if cls._has_key_nested(values, field):
+                raise ValueError(f"{field} can't be user-defined; set internally")
+        return values
+
+
+class FrozenModel(CoreBaseModel):
     """
+    Inherits all of the configuration from `CoreBaseModel`.
 
-    model_config = ConfigDict(extra="forbid", populate_by_name=True, frozen=True)
+    The following overrides are applied:
+    - extra="forbid": Forbids additional fields beyond those defined in the model.
+    - frozen=True: Disallows modification of field values after object instantiation.
 
+    This model is intended for cases where additional fields are not permitted and the
+    object data must remain immutable after initialization.
+    """
+
+    model_config = ConfigDict(extra="forbid", frozen=True)
 
-class PermissiveFrozenModel(PrettyBaseModel):
-    """Pydantic model with a customized ``ConfigDict`` configurations for GeoIPS.
 
-    This model extends ``PrettyBaseModel`` and uses Pydantic's ConfigDict to provide
-    customized configurations. It is intended for use in cases where additional fields
-    are allowed, but the object data cannot be modified after initialization.
+class PermissiveFrozenModel(CoreBaseModel):
     """
+    Inherits all of the configuration from `CoreBaseModel`.
 
-    model_config = ConfigDict(extra="allow", populate_by_name=True, frozen=True)
+    The following overrides are applied:
+    - extra="allow": Allows additional fields beyond those defined in the model.
+    - frozen=True: Disallows modification of field values after object instantiation.
 
+    This model is intended for cases where additional fields are permitted and the
+    object data must remain immutable after initialization.
+    """
 
-class DynamicModel(PrettyBaseModel):
-    """Pydantic model with a customized ``ConfigDict`` configurations for GeoIPS.
+    model_config = ConfigDict(extra="allow", frozen=True)
```

Comment:
Was corebasemodel just set up in order to standardize using more of the available configdict options for these different types of models (previously they just used the extra and the frozen options? )

And populate by name has been removed from these,  since it is now defaulted in the core base model?